### PR TITLE
Full language support

### PR DESCRIPTION
**Pull request by TommYDeeee**
Full language support WiP

✅ Subset 
✅ Rule modifiers 
✅ Rule tags 
✅ Imports, Includes
✅ Meta sections 
✅ Pattern modifiers 
✅ Hex patterns
✅  Regex patterns
Full expression support: 
✅  Missing operators in Pratt Parser (DIV, MULT, GE, LT,....) 
✅  Filesize, entrypoint, defined 
✅  Bitewise operators 
✅ Literals – float, integer, strings, regexep, utf8 string lit 
✅ Pattern length, pattern offset, pattern count 
✅ Function call expressions
✅ Indexing expressions 
✅ Iterators (range, expression tuple,...) 
✅ Of expression 
✅ For expression 
✅ Anchors (AT, IN) 
✅ Quantifiers (ALL, ANY, NONE) 
✅ Tuples (identifiers, identifier wildcards)

### CODE REVIEW

File: yara.ungram [hunk 1]
```diff
@@ -6,9 +6,21 @@ Rule =
...
+        Meta? Strings? Condition
     '}'
 
+Meta = 
+    'meta' ':'
+        MetaStmt*
+
+MetaStmt =
+    'identifier' '=' 
+        |  'true'
```
Comment (MatejKastak) [inline yara.ungram]: It seems a bit inconsistent to have other types of literals, but for bool have values explicitly listed here.

I am not pushing to change it, but have you considered to have `'bool_lit'` here?

Comment (TommYDeeee) [reply yara.ungram]: Yes, thank you! This is from subset where there was no point in trying to make it consistant as there were no 'string_lit' or 'int_lit' values. I will unify it.

Comment (TommYDeeee) [reply yara.ungram]: Fixed in latest commit together with hex string support. Thank you again for a review

File: tests/test17.in [hunk 2]
```diff
@@ -0,0 +1,7 @@
+rule test
+{
+	strings:
+		$a = /baz./is
```
Comment (MatejKastak) [inline tests/test17.in:4]: I remember fighting a lot with regexps that contain escaped `\/` inside.

I recommend that you add a test for that also. There are a lot of regexes for web URLs that contain it.

For example:
```
$a = /http(s)?:\/\/example\.org\//
```

File: yara.ungram [hunk 3]
```diff
@@ -39,12 +38,44 @@ Strings =
         VariableStmt*
 
 VariableStmt =
-    'variable' '=' Pattern
+    'variable' '=' Pattern
```
Comment (MatejKastak) [inline yara.ungram]: Nit: whitespace
```suggestion
    'variable' '=' Pattern
```

**Review by MatejKastak — APPROVED** [review]

File: tests/test17.in (hunk 2 above)
Comment (TommYDeeee) [reply tests/test17.in:4]: Thanks! It indeed did not parse this regex correctly. I have fixed it and it should be ok now. Also added test for this

**Review by MatejKastak — APPROVED** [review]

**Comment by MatejKastak** [comment]
I played with the code a bit.

It would be nice if we provided a similiar binary for testing to dump AST to stdout.

Simple bin.rs or main.rs
```rust
use std::fs;

use yara_parser::SourceFile;

fn main() {
    let path = std::env::args().nth(1).expect("No arguments provided");
    let input =
        fs::read_to_string(&path).unwrap_or_else(|_| panic!("Failed to read input file {}", path));
    let ast_struct = SourceFile::parse(&input);
    println!("{:#?}", ast_struct.tree().syntax);
}
```

**Comment by TommYDeeee** [comment]
> It would be nice if we provided a similar binary for testing to dump AST to stdout.

Oh sure, I have almost the same binary locally. I just didn't push it here. I will do it, thanks

**Comment by TommYDeeee** [comment]
> It would be nice if we provided a similiar binary for testing to dump AST to stdout.

Resolved in https://github.com/avast/yara-x-parser/pull/4/commits/5dde9f840ebdae2129c33bb11cb65a8c2e2cfc97

**Review by MatejKastak — APPROVED** [review]

**Review by MatejKastak — APPROVED** [review]

**Review by MatejKastak — APPROVED** [review]
LGTM :+1:

**Pull request merged**